data/state/accountsDB.go

```diff
@@ -546,14 +562,50 @@ func (adb *AccountsDB) RecreateTrie(rootHash []byte) error {
 	if err != nil {
 		return err
 	}
-	if newTrie == nil {
+	if check.IfNil(newTrie) {
 		return ErrNilTrie
 	}
 
 	adb.mainTrie = newTrie
 	return nil
 }
 
+// RecreateAllTries recreates all the tries from the accounts DB
+func (adb *AccountsDB) RecreateAllTries(rootHash []byte) (map[string]data.Trie, error) {
+	recreatedTrie, err := adb.mainTrie.Recreate(rootHash)
+	if err != nil {
+		return nil, err
+	}
+
+	leafs, err := recreatedTrie.GetAllLeaves()
+	if err != nil {
+		return nil, err
+	}
+
+	allTries := make(map[string]data.Trie)
+	allTries[string(rootHash)] = recreatedTrie
+
+	for _, leaf := range leafs {
+		account := &Account{}
+		err = adb.marshalizer.Unmarshal(account, leaf)
+		if err != nil {
+			log.Trace("this must be a leaf with code", "err", err)
```

Comment:
Code is saved as a []byte in the leaf. When accessing the leaf there is no information whether it is a code / account. So first there is a try to unmarshal into an account, otherwise it must be code.